Neos.Flow/Classes/Cache/CacheManager.php

```diff
@@ -177,19 +178,24 @@ public function registerCache(FrontendInterface $cache, bool $persistent = false
     /**
      * Returns the cache specified by $identifier
      *
+     * @template T of object
      * @param string $identifier Identifies which cache to return
-     * @return FrontendInterface The specified cache frontend
+     * @param class-string<T>|null $className The expected CacheManager for the cache frontend
+     * @return (T is null ? LowLevelFrontendInterface : T && LowLevelFrontendInterface) The specified cache frontend
```

Comment:
I still struggle a bit with the syntax for return types. PHPStan seems to understand that but it may be a semi-geniuis  idea alltogether.

Reply:
Hmm i see your point. Maybe its better to delay this discussion in a separate pr as this is a general problem we face in many places see FusionRuntime but also the ValueAccessor and we should discuss our philosophy once in an issue and draw out a path for all parts that would profit from this solution. Otherwise i fear this discussion might take over in this pr and scatter our focus from the important parts ;)

Reply:
@mhsdesign done as suggested